### PR TITLE
AMS refactor

### PR DESCRIPTION
**Pull request by tofu-rocketry**
This is a pull request back to GC's branch as I'd like to have feedback on my changes. I've attempted to make the code for AMS support a bit simpler and easier for the user.

- Collapse options to switch between STOMP message brokers and AMS to a single config option to simplify things for the user.
- Rename AMS project config option to 'ams_project' to make purpose clear. This also means that it doens't need to go into a completely separate AMS section.
- Remove unnecessary comments from receiver config.
- Update host & port for broker in receiver config (not done for sender to avoid misconfiguration at a site cuasing problems centrally).
- Collapse destination_type into protocol to simplify code. `MQ-BROKER` converts to `STOMP`, and `HTTPS` converts to `AMS`.
- Change name of section that variables are fetched form the config to match new names.
- Remove some unnecessary checks and reorder arguments.
- Remove unncessary `try...except` that raises the exception without modifying it.
- Replace magic strings used for protocols with constants defined in the Ssm2 class as this is good practice.
- Merge `_pull_msg_rest_ams` method into `pull_msg_rest`. Now that the `pull_msg_rest` method has been simplified, the split with the `..._ams` method is a bit unnecessary.
- Add separate token argument to Ssm2 to remove overloading of password
argument as password and username seem redundant and could be removed at
a later stage.

### CODE REVIEW

File: bin/receiver.py [hunk 1]
```diff
@@ -104,36 +104,26 @@ def main():
...
-    # Set defaults for MQ-BROKER only variables
-    use_ssl = None
-    # Set defaults for AMS only variables
-    project = None
-    token = ""
-
     log.info(LOG_BREAK)
     log.info('Starting receiving SSM version %s.%s.%s.', *__version__)
 
     # Determine the protocol and destination type of the SSM to configure.
```
Comment (gregcorbett) [inline bin/receiver.py]: as there is no longer a destination type, this comment should be reworded

**Comment by gregcorbett** [comment]
> Update host & port for broker in receiver config (not done for sender to avoid misconfiguration at a site cuasing problems centrally).

It looks like you've updated host and port in the sender config and not in the receiver config?

More generally, do we want to "specify" a host and port in either config at all? It changes between sending via STOMP or via AMS, so might get confusing. Or at least a comment saying the details are for a specific means of sending as per the bdii and network settings?

**Comment by gregcorbett** [comment]
The AMS documentation in README.md and migrating_to_ams_broker.md should be updated to reflect the changes in the config files.

**Comment by tofu-rocketry** [comment]
Fair points. I've addressed them all.

File: bin/receiver.py (hunk 1 above)
Comment (tofu-rocketry) [reply bin/receiver.py]: Done.

**Pull request merged**